### PR TITLE
More ssh

### PR DESCRIPTION
**Pull request by mrocklin**
Filled out some conversions to/from `Temp(SSH(o))` to enable smooth transfer via the remote host.

```Python
In [1]: from blaze import Data, into

In [2]: auth = {'hostname': '54.91.57.226',
        'key_filename': '/home/mrocklin/.ssh/cdh_testing.key',
        'username': 'ubuntu'}

In [3]: t = into('hive://54.91.57.226/default::iris_3', 'data/iris.csv', **auth)

In [4]: t
Out[4]: Table('iris_3', MetaData(bind=Engine(hive://hdfs@54.91.57.226:10000/default)), Column('sepal_length', Float(), table=<iris_3>), Column('sepal_width', Float(), table=<iris_3>), Column('petal_length', Float(), table=<iris_3>), Column('petal_width', Float(), table=<iris_3>), Column('species', String(), table=<iris_3>), schema=None)

In [5]: Data(t)
Out[5]: 
    sepal_length  sepal_width  petal_length  petal_width      species
0            5.1          3.5           1.4          0.2  Iris-setosa
1            4.9          3.0           1.4          0.2  Iris-setosa
2            4.7          3.2           1.3          0.2  Iris-setosa
3            4.6          3.1           1.5          0.2  Iris-setosa
4            5.0          3.6           1.4          0.2  Iris-setosa
5            5.4          3.9           1.7          0.4  Iris-setosa
6            4.6          3.4           1.4          0.3  Iris-setosa
7            5.0          3.4           1.5          0.2  Iris-setosa
8            4.4          2.9           1.4          0.2  Iris-setosa
9            4.9          3.1           1.5          0.1  Iris-setosa
...
```

### CODE REVIEW

File: into/backends/ssh.py [hunk 1]
```diff
@@ -206,7 +206,7 @@ def file_to_temp_ssh_file(typ, data, **kwargs):
     """ Generic convert function sending data to ssh(data)
 
     Needs to be partially evaluated with a type"""
-    fn = '.%s.%s' % (uuid.uuid1(), typ.canonical_extension)
+    fn = '%s.%s' % (uuid.uuid1(), typ.canonical_extension)
```
Comment (cpcloud) [inline into/backends/ssh.py]: is per the conversation with you and @quasiben? maybe a small comment stating why we can't hide this file.

File: into/utils.py [hunk 2]
```diff
@@ -204,5 +204,15 @@ def ignoring(*exceptions):
         pass
 
 
+def into_path(*path):
+    """ Path to file in into directory
+
+    >>> into_path('backends', 'tests', 'myfile.csv')  # doctest: +SKIP
+    '/home/user/into/into/backends/tests/myfile.csv')
```
Comment (cpcloud) [inline into/utils.py]: small typo here, mismatched paren on the end

File: into/backends/ssh.py [hunk 3]
```diff
@@ -219,7 +219,20 @@ def file_to_temp_ssh_file(typ, data, **kwargs):
...
     return append(target, data, **kwargs)
+
+
+@convert.register(Temp(SSH(TextFile)), (TextFile, Temp(TextFile)))
+@convert.register(Temp(SSH(JSONLines)), (JSONLines, Temp(JSONLines)))
+@convert.register(Temp(SSH(JSON)), (JSON, Temp(JSON)))
+@convert.register(Temp(SSH(CSV)), (CSV, Temp(CSV)))
+def file_to_temp_ssh_file(data, **kwargs):
+    fn = '%s' % uuid.uuid1()
+    if isinstance(data, _Temp):
```
Comment (cpcloud) [inline into/backends/ssh.py]: i've been using `getattr(data, 'persistent_type', type(data))` for this. just a comment really, no action needed

**Comment by cpcloud** [comment]
looks good to me, few minor comments

File: into/backends/ssh.py (hunk 3 above)
Comment (mrocklin) [inline into/backends/ssh.py]: I considered that.  I was a bit confused by the code that I was writing so I decided to stick to fully explicit on this one.

File: into/utils.py (hunk 2 above)
Comment (mrocklin) [inline into/utils.py]: Fixed

File: into/backends/ssh.py (hunk 1 above)
Comment (mrocklin) [inline into/backends/ssh.py]: added comment

**Comment by mrocklin** [comment]
Merging soon if no comments

**Pull request merged**